### PR TITLE
LGA-1046 - Reset the task last message at the start of task.run

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Reset the task last message at the start of task.run
Added `Check passphrase and try again` message when diversity passphrase is incorrect

## Any other changes that would benefit highlighting?

Currently the OBIEE extracts seem to failing with the error message `An error occurred creating the zip file` however the extract seems to be downloadable.

https://docs.celeryproject.org/en/3.0/userguide/tasks.html#instantiation states that:
> A task is not instantiated for every request, but is registered in the task registry as a global instance.

So this means that the same instance of the OBIEEExportTask is being used to handle multiple exports which means unless explicitly reset instance properties such as `message` retain their values through multiple exports.

The current issue is cause by when one export fails due to a legitimate error such as invalid passphrase, all subsequent successful exports will have the same message because `self.message` is only set when there is an error.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Comment by exonian** [comment]
That coverage... 😭

File: cla_backend/apps/reports/tests/test_tasks.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with self.assertRaisesMessage(InternalError, "Wrong key or corrupt data"):
+            self.run_task("badpass")
+        self.assertEqual("Check passphrase and try again", self.task.message)
+
+    def test_message_reset_per_run(self):
+        with self.assertRaisesMessage(InternalError, "Wrong key or corrupt data"):
+            self.run_task("badpass")
+
+        self.assertEqual("Check passphrase and try again", self.task.message)
+        self.run_task("cla")
```
Comment (exonian) [inline cla_backend/apps/reports/tests/test_tasks.py]: Where is this correct password defined?

Comment (said-moj) [reply cla_backend/apps/reports/tests/test_tasks.py]: Update method with the following now:
`        # Run task with an invalid passphrase
        with self.assertRaisesMessage(InternalError, "Wrong key or corrupt data"):
            self.run_task("badpass")
        self.assertEqual("Check passphrase and try again", self.task.message)

        # Run task with the correct passphrase
        self.run_task("cla")
        self.assertEqual("", self.task.message)`

Correct passphrase for testing is "cla"

Comment (exonian) [reply cla_backend/apps/reports/tests/test_tasks.py]: I guessed that but I can't work out *why* it's correct

Comment (said-moj) [reply cla_backend/apps/reports/tests/test_tasks.py]: The diversity data is encrypted using public/private key mechanism, the development keys are stored in https://github.com/ministryofjustice/cla_backend/tree/master/cla_backend/apps/legalaid/utils/keys (provided as environment variables in other environments)

The private key requires a passphrase which for development is cla

Comment (exonian) [reply cla_backend/apps/reports/tests/test_tasks.py]: Ah super, thanks

**Pull request merged**